packages/web-components/src/components/rux-tree-node/rux-tree-node.tsx

```diff
@@ -253,34 +256,56 @@ export class RuxTreeNode {
     }
 
     render() {
-        const attrs = this._hasChildren && { role: 'group' }
+        const {
+            componentId,
+            expanded,
+            hasChildren,
+            hasPrefix,
+            hasSuffix,
+            iconName,
+            selected,
+        } = this
+        const attrs = hasChildren && { role: 'group' }
 
         return (
             <Host
                 role="treeitem"
-                aria-expanded={this.expanded ? 'true' : 'false'}
-                aria-selected={this.selected ? 'true' : 'false'}
+                aria-expanded={expanded ? 'true' : 'false'}
+                aria-selected={selected ? 'true' : 'false'}
                 onClick={(event: MouseEvent) =>
                     this._handleTreeNodeClick(event)
                 }
             >
                 <div
-                    id={this.componentId}
+                    id={componentId}
                     class={{
                         'tree-node': true,
-                        'tree-node--expanded': this.expanded,
-                        'tree-node--has-children': this._hasChildren,
-                        'tree-node--selected': this.selected,
+                        'tree-node--expanded': expanded,
+                        'tree-node--has-children': hasChildren,
                     }}
                 >
                     <div class="parent" tabindex="0">
-                        {this._hasChildren && (
-                            <i
-                                onClick={(e) => this._handleArrowClick(e)}
+                        {hasChildren && (
+                            <rux-icon
                                 class="arrow"
-                            ></i>
+                                onClick={this._handleArrowClick.bind(this)}
+                                size="1.25rem"
+                                icon={iconName}
+                            />
+                        )}
+                        {hasPrefix && (
+                            <span class="prefix">
+                                <slot name="prefix"></slot>
```

Comment:
since the slot is being conditionally rendered, we can't add slotted content after the component renders. 

For example, we should be able to do: 

```
<body>
        <rux-tree-node style="width: 300px;">whatup</rux-tree-node>
        <button id="add">add a prefix</button>
        <script>
            add.addEventListener('click', () => {
                const el = document.querySelector('rux-tree-node')
                const icon = document.createElement('rux-status')
                icon.setAttribute('status', 'critical')
                icon.setAttribute('slot', 'prefix')
                el.appendChild(icon)
            })
        </script>
    </body>
```